brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java

```diff
@@ -109,10 +115,76 @@ private boolean shouldMergePublications(Publication publication) throws NotFound
         var cristinIdentifier = getCristinIdentifier(publication);
         if (nonNull(cristinIdentifier)) {
             publicationsToMerge = getPublicationsByCristinIdentifier(cristinIdentifier);
-            return !publicationsToMerge.isEmpty();
+            boolean shouldMerge = !publicationsToMerge.isEmpty();
+            source = shouldMerge ? MergeSource.CRISTIN : MergeSource.NOT_RELEVANT;
+            return shouldMerge;
         }
+        if (hasDoi(publication)) {
+            var shouldMerge= existingPublicationHasSameDoi(publication);
+            source = shouldMerge ? MergeSource.DOI : MergeSource.NOT_RELEVANT;
+            return shouldMerge;
+        } else {
+            var shouldMerge = existingPublicationHasSamePublicationContent(publication);
+            source = shouldMerge ? MergeSource.SEARCH : MergeSource.NOT_RELEVANT;
+            return shouldMerge;
+        }
+    }
+
+    private boolean existingPublicationHasSamePublicationContent(Publication publication) {
+        publicationsToMerge = searchForPublicationsByTypeAndTitle(publication);
+        return !publicationsToMerge.isEmpty();
+
+    }
+
+    private List<Publication> searchForPublicationsByTypeAndTitle(Publication publication) {
+        var response = fetchResponse(searchByTypeAndTitleUri(publication));
+        return response.map(this::toResponse)
```

Comment:
Not unsurprisingly, there is a likelihood that this will miss more often than not. I'd consider a different approach with a similarity/distance metric between similar titles and for example (weirdly) the number of pages

Reply:
Type is good though